tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1346,11 +1384,22 @@ def check_parameters(self, node):
             if param not in docparams:
                 missing_params.append(param)
 
+        # check that all keyword-only args are documented
+        missing_kwonly_args = []
+        for param in keyword_only_args:
+            if param not in keyword_args:
+                missing_kwonly_args.append(param)
```

Comment:
Correct me if I'm confused - don't we want to check two things here? 

1) all method kw are described in the docstrings
2) all docstring kw are present in the method signature

If I understand the code right, we're only checking 1 here.

Reply:
ah missed that -- will update

Reply:
used ^ of 2 sets

Reply:
Should we have different messages for 1 and 2? Right now we say

> "%s" found as keyword in docstring, not in method signature. :keyword: arguments should be keyword-only arguments in method signature. See details

Which might be confusing in scenario 1 since it's swapped - found in the method signature, not in the docstring.

Reply:
'"%s" missing in docstring or in method signature. There should be a direct correlation between :keyword: arguments in the docstring and keyword-only arguments in method signature. See details: '
            'https://azure.github.io/azure-sdk/python_documentation.html#docstrings',
            "docstring-keyword-should-match-keyword-only",
            "Docstring keyword arguments and keyword-only method arguments should match.",

Reply:
what if we use `match`?

Reply:
Works for me